cedar-policy-core/src/entities.rs

```diff
@@ -823,34 +990,201 @@ mod json_parsing_tests {
+            "parents": [
+                { "__entity": { "type" : "test_entity_type", "id" : "bob"} },
+                { "__entity": { "type": "test_entity_type", "id": "catherine" } }
+            ]
+        }
+        ]);
+        let eparser: EntityJsonParser<'_> =
+            EntityJsonParser::new(None, Extensions::all_available(), TCComputation::ComputeNow);
+        let error = eparser.from_json_value(json).err().unwrap().to_string();
+        assert!(
+            error.contains("in uid field of <unknown entity>, expected a literal entity reference"),
+            "{}",
+            error
+        );
+    }
+
+    #[test]
+    fn no_expr_escapes3() {
+        let json = serde_json::json!(
+        [
+        {
+            "uid" : {
+                "type" : "test_entity_type",
+                "id" : "Alice"
+            },
+            "attrs": {
+                "bacon": "eggs",
+                "pancakes": { "__expr" : "[1,2,3]" },
+                "waffles": { "key": "value" },
+                "toast" : { "__extn" : { "fn" : "decimal", "arg" : "33.47" }},
+                "12345": { "__entity": { "type": "test_entity_type", "id": "bob" } },
+                "a b c": { "__extn": { "fn": "ip", "arg": "222.222.222.0/24" } }
+            },
+            "parents": [
+                { "__entity": { "type" : "test_entity_type", "id" : "bob"} },
+                { "__entity": { "type": "test_entity_type", "id": "catherine" } }
+            ]
+        }
+        ]);
+        let eparser: EntityJsonParser<'_> =
+            EntityJsonParser::new(None, Extensions::all_available(), TCComputation::ComputeNow);
+        let error = eparser.from_json_value(json).err().unwrap().to_string();
+        assert!(
+            error.contains("`__expr` tag is no longer supported"),
+            "Actual error message was: {}",
+            error
+        );
+    }
+
+    #[test]
+    fn no_expr_escapes4() {
+        let json = serde_json::json!(
+        [
+        {
+            "uid" : {
+                "type" : "test_entity_type",
+                "id" : "Alice"
+            },
+            "attrs": {
+                "bacon": "eggs",
+                "waffles": { "key": "value" },
+                "12345": { "__entity": { "type": "test_entity_type", "id": "bob" } },
+                "a b c": { "__extn": { "fn": "ip", "arg": "222.222.222.0/24" } }
+            },
+            "parents": [
+                { "__expr": { "type" : "test_entity_type", "id" : "bob"} },
+                { "__entity": { "type": "test_entity_type", "id": "catherine" } }
+            ]
+        }
+        ]);
+        let eparser: EntityJsonParser<'_> =
+            EntityJsonParser::new(None, Extensions::all_available(), TCComputation::ComputeNow);
+        let error = eparser.from_json_value(json).err().unwrap().to_string();
+        assert!(
+            error.contains(r#"in parents field of `test_entity_type::"Alice"`, expected a literal entity reference"#),
```

Comment:
it would be better to have a message explicitly mentioning the `__expr` escape in this case too